controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -307,10 +309,40 @@ public String getScopeName() {
                             long scaleTs = record.getLeft();
                             CompletableFuture<List<Segment>> list = Futures.allOfWithResults(
                                     record.getRight().stream().map(this::getSegment)
-                                    .collect(Collectors.toList()));
+                                            .collect(Collectors.toList()));
+
+                            return list.thenApply(segments -> new ImmutablePair<>(scaleTs, segments));
+                        }).collect(Collectors.toList())))
+                .thenApply(scalePair -> {
```

Comment:
For organization purposes, it would be better to have this logic implemented in a private method like before.

Reply:
i will move it to a private method.. though earlier this logic not there. a different cumulative split and merge count computation was there but it was made public and exposed on the store interface.. which was redundant and appears to have  only been done for testing purposes.